lib/BackgroundJobs/NotifyUserTask.php

```diff
@@ -0,0 +1,197 @@
+        $this->mailer = $mailer;
+        $this->defaults = $defaults;
+        $this->userManager = $userManager;
+        $this->l10n = $l10n;
+        $this->groupManager = $groupManager;
+
+        $connection = \OC::$server->getDatabaseConnection();
+        $this->userDAO = new UserDAO($connection);
+    }
+
+    /**
+     * @param $argument
+     */
+    protected function run($argument) {
+        if ($this->config->getExpirationDays() > 0 && $this->config->getExpirationMailDaysBefore() > 0) {
+
+            $daysTillNotification = $this->config->getExpirationDays() - $this->config->getExpirationMailDaysBefore();
+
+            $users = $this->userDAO->findAllUsersExpirationDataBy('notification_sent', 0);
+
+            foreach ($users as $user) {
+                $passwordLastChanged = $user['last_changed'];
+                $userObject = $this->userManager->get($user['uid']);
+
+                if (strtotime('+' . $daysTillNotification . ' days', $passwordLastChanged) <= time()
+                    && $this->userPasswordCanExpire($userObject)) {
+
+                    $result = $this->sendNotificationMail($userObject, $passwordLastChanged);
+
+                    if ($result == true) {
+                        $this->setNotificationSend($user['uid']);
+                    } else {
+                        $this->logger->debug('Sending email to user "' . $user['uid'] . '" with email "' . $userObject->getEMailAddress() .'" failed');
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param IUser $userObject
+     * @param integer $passwordLastChanged
+     *
+     * @return bool
+     */
+    protected function sendNotificationMail($userObject, $passwordLastChanged) {
+        $userId = $userObject->getUID();
+        $userEmail = $userObject->getEMailAddress();
+
+        if (strlen($userEmail) > 1 && filter_var($userEmail, FILTER_VALIDATE_EMAIL)) {
+            $template_var = [
+                'user' => $userId,
+                'linkToPage' => $this->config->getNextcloudHost(),
+                'expireDay' => date('j. F Y', strtotime('+' . $this->config->getExpirationDays() . ' days', $passwordLastChanged))
+            ];
+
+            $html_template = new TemplateResponse('password_policy', 'email.usernotify_html', $template_var, 'blank');
+            $html_part = $html_template->render();
+
+            $plaintext_template = new TemplateResponse('password_policy', 'email.usernotify_plaintext', $template_var, 'blank');
+            $plaintext_part = $plaintext_template->render();
+
+            $subject = $this->l10n->t('Your password is about to expire.');
+            $from = Util::getDefaultEmailAddress('register');
+
+            $message = $this->mailer->createMessage();
+            $message->setFrom([$from => $this->defaults->getName()]);
+            $message->setTo([$userEmail => $userId]);
+            $message->setSubject($subject);
```

Comment:
In the meantime setTemplate was added, which comes with nice and easy HTML emails.
Maybe you could also have a look at this.